### PR TITLE
Collect statistics for the kube object namespace.

### LINKED ISSUE #3: Collect stats on namespaces
- Number of namespaces
- Histogram or average age of namespaces
- Histogram or average number of (pods, services, deployments, jobs) per namespace

### PR DESCRIPTION
**Pull request by kenan435**
Added several statistics for the k8s namespace object.

fixes #3

### CODE REVIEW

File: pkg/report/record.go [hunk 1]
```diff
@@ -32,6 +32,23 @@ type Record struct {
 	MasterVersion *string `json:"masterVersion,omitempty"`
 	// Nodes is a list of node-specific information from the reporting cluster.
 	Nodes []Node `json:"nodes,omitempty"`
+	// Namespace contains statistics regarding the k8s namespace objects
+	Namespace Namespace `json:"namespaces,omitempty"`
+}
+
+type Namespace struct {
+	// Total number of active namespaces in a k8s cluster.
+	TotalNSSum float64 `json:"total"`
```
Comment (philips) [inline pkg/report/record.go]: Total and Sum seems redundant.

**Review by philips — COMMENTED** [review]
Could you update the README too to provide an exmaple?

File: .gitignore [hunk 2]
```diff
@@ -2,3 +2,4 @@
 /.go
 /.push-*
 /.container-*
+.idea/
```
Comment (thockin) [inline .gitignore]: What is this about?

File: pkg/report/record.go (hunk 1 above)
Comment (thockin) [reply pkg/report/record.go]: Agree, also please don't float for things like total.  int32 should suffice.

File: pkg/report/record.go [hunk 3]
```diff
@@ -32,6 +32,23 @@ type Record struct {
 	MasterVersion *string `json:"masterVersion,omitempty"`
 	// Nodes is a list of node-specific information from the reporting cluster.
 	Nodes []Node `json:"nodes,omitempty"`
+	// Namespace contains statistics regarding the k8s namespace objects
+	Namespace Namespace `json:"namespaces,omitempty"`
+}
+
+type Namespace struct {
```
Comment (thockin) [inline pkg/report/record.go]: Call this NamespaceStats - it doesn't represent a Namespace like a Node struct does.

File: pkg/report/record.go [hunk 4]
```diff
@@ -32,6 +32,23 @@ type Record struct {
...
+	// Total number of active namespaces in a k8s cluster.
+	TotalNSSum float64 `json:"total"`
+	// The average age of active namespaces in a k8s cluster.
+	LifetimeNSAvg float64 `json:"namespaceLifetimeAvg"`
```
Comment (thockin) [inline pkg/report/record.go]: Don't need "NS" or "namespace" on this field

**Review by thockin — COMMENTED** [review]
I was thinking histogram or some struct `{min, p25, p50, p75, p90, p99, max, mean}` would be a better way to represent numeric distributions.  I am not a numerics guy, so open to feedback.  Short of raw numbers, what is the best way to collect stats that aggregate properly?

**Comment by thockin** [comment]
https://godoc.org/google.golang.org/grpc/benchmark/stats

https://godoc.org/github.com/prometheus/client_golang/prometheus#Histogram

File: .gitignore (hunk 2 above)
Comment (kenan435) [reply .gitignore]: left over from the webstorm IDE. reverted

File: pkg/report/record.go (hunk 3 above)
Comment (kenan435) [reply pkg/report/record.go]: done

File: pkg/report/record.go (hunk 4 above)
Comment (kenan435) [reply pkg/report/record.go]: done

File: pkg/report/record.go (hunk 1 above)
Comment (kenan435) [reply pkg/report/record.go]: Readme updated.

File: pkg/report/record.go (hunk 4 above)
Comment (kenan435) [reply pkg/report/record.go]: @thockin Statistics are not my strongest suit either so we are all set :) Anyhow, I added a sample .json output to the Readme file for pods and lifetime stats. Can you have a look and let me know what you think? I am embedding a histogram into each namespace stat (jobs, svc, deply, etc..).

**Comment by thockin** [comment]
This fell off the bottom of my email.  For the first time in months I can see the bottom (and there's a lot there).  Is this something we want to bring back to life?

**Pull request closed without merging**